pulsar-client/src/main/java/org/apache/pulsar/client/impl/auth/oauth2/ClientCredentialsFlow.java

```diff
@@ -137,19 +138,18 @@ public static ClientCredentialsFlow fromParameters(Map<String, String> params) {
      * @return
      * @throws IOException
      */
-    private static KeyFile loadPrivateKey(String privateKeyURL) throws IOException {
+    @VisibleForTesting
+    static KeyFile loadPrivateKey(String privateKeyURL) throws IOException {
         try {
             URLConnection urlConnection = new org.apache.pulsar.client.api.url.URL(privateKeyURL).openConnection();
             try {
                 String protocol = urlConnection.getURL().getProtocol();
-                String contentType = urlConnection.getContentType();
-                if ("data".equals(protocol) && !"application/json".equals(contentType)) {
+                if ("data".equals(protocol) && !"application/json".equals(urlConnection.getContentType())) {
```

Comment:
> URL content type is only required if the scheme is data

But we already only verify the `contentType` if the protocol is "data"?

I don't see the different for this point before and after this patch.

Reply:
The change here does not make any difference to the correctness. but only avoid calling `getContentType()` when the protocol is "data".

Reply:
Indeed